### DISCUSSION TITLE
[Feature] Add AirPlay to iOS app

### DISCUSSION THREAD

**Discussion by sidamos** [immich-app/immich#9267] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Please add share via AirPlay to the iOS app. Thanks!

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by WonderboyFromMars** [comment]
This feature is actually a must have on iOS devices! Especially with a missing Immich Apple TV app.
AirPlay is an essential function and de facto standard on apple systems. We use it regularly to stream videos and pictures to our Apple TV

**Comment by lifecounter** [comment]
I believe this issue warrants more attention. AirPlay integration should be one of the highest priorities for the Immich app. This isn’t just about compatibility with the Apple TV—since many other devices support AirPlay—it’s fundamentally about meeting the needs of the vast iPhone user base.

Currently, the default iOS Photos app has become the go-to solution for displaying holiday photos and videos on a TV. While this setup works well for recent memories, it falls short when it comes to accessing older media stored in Immich. In such cases, users are forced to view their cherished moments on an iPhone, PC, or laptop—a solution that hardly does justice to the experience of reminiscing on a big screen.

Given the increasing demand for seamless streaming across devices, integrating native AirPlay support into Immich would significantly enhance its usability. With this feature, users could effortlessly stream their photos and videos from Immich to any AirPlay-enabled device, transforming the way they enjoy and share their memories. Not only would this meet modern user expectations, but it would also solidify Immich’s position as a versatile and competitive media management solution.

Prioritizing AirPlay integration is essential. It would bridge the gap between the convenience of a dedicated app like Immich and the superior viewing experience that comes with streaming to larger screens, ultimately making it easier for users to relive and share their most treasured moments. An Apple TV app would be a solution, but it would not be THE solution to this problem. 

Immich developers: If not already done, try it. Use the photos app and AirPlay your Photos and Videos to an Apple TV. You would support my/our position, as soon as you have tried. It´s the best possible user experience for this problem.

**Reply by bo0tzz** [reply]
> AirPlay integration should be one of the highest priorities for the Immich app.

Frankly, priorities are not for you to determine.

**Reply by petersuba** [reply]
Wow. Frankly, why have a forum allowing people to express their opinions (not forcing it on the developers) if they get this attitude from staff. You have a popular solution that is used by many people. They will all have their own perspectives and thoughts. I did not think this needed to be explained to someone who works on this solution, even if I understand the pressure that comes with the popularity, the breadth and diversity of proposed new or changed features and bugs from the community. 

I have just tried to do exactly what the OP is describing the other day on an LG TV that supports AirPlay with Immich for the family and was disappointed that I could not. I wish it existed. I don't think this is essential relative to many other things, but I also would be happy to see it happen and can't understand this reaction to the proposal.

**Reply by bo0tzz** [reply]
To be clear, people are absolutely welcome to share their opinions and express that they want a particular feature, of course. I agree that my response above was overly harsh but it gets pretty old to have people telling us that their particular wanted features should be our number one priority all the time.

**Reply by christerdk** [reply]
Welcome to software development, bo0tzz, it’s been like this since the very dawn of software development, for good or worse.

**Comment by jonvdveen** [comment]
Would love to see this. I've been using the [Immich Gallery App](https://github.com/mensadilabs/Immich-Gallery) on Apple TV. I'm pretty impressed, despite some rough edges. But when I'm at someone else's home, it would be great to cast images or a slideshow to their Apple TV without mirroring my device's screen.
